### PR TITLE
[GEOT-6008]: Supporting multilevel sidecar files

### PR DESCRIPTION
**Pull request by dromagnoli**

### CODE REVIEW

File: modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIProviderFactory.java [hunk 1]
```diff
@@ -71,7 +90,7 @@ public static MultiLevelROIProvider createFootprintProvider(File referenceFile)
      * @throws Exception
      */
     public static MultiLevelROIProvider createFootprintProvider(
-            final File referenceFile, Geometry granuleBounds) {
+            final File referenceFile, Geometry granuleBounds, Hints hints) {
```
Comment (aaime) [inline modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIProviderFactory.java]: The hints are not really used here, probably a leftover from a failed experiment?

Comment (dromagnoli) [reply modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIProviderFactory.java]: mmm nope. 
this implementation do not use hints, while the MultiLevelROIProviderMosaicFactory is actually using it

Comment (aaime) [reply modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIProviderFactory.java]: But it's a static method, there is no notion of override...

**Comment by dromagnoli** [comment]
Indeed my intention was to update the GeoServer doc too with new properties and usage. 
I will do it right now and refer it here

File: modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIProviderFactory.java (hunk 1 above)
Comment (dromagnoli) [reply modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIProviderFactory.java]: Ah! Right... nope... no reflection. I think that at the very beginning I was having issues with the MultiLevelROIProviderMosaicFactory subclass methods. 
I'm going to remove it, thanks.

**Comment by dromagnoli** [comment]
PR for GeoServer documentation update, ready:
https://github.com/geoserver/geoserver/pull/2876

**Pull request merged**